### PR TITLE
gromacs 2019.3

### PR DESCRIPTION
**Pull request by YoshitakaMo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Update to Gromacs 2019.3. In this formula, gcc-8 compiler was used because gcc-9 and Apple LLVM could not compile successfully.

I also fix an error on detecting CPU. see https://redmine.gromacs.org/issues/2927

### CODE REVIEW

**Comment by fxcoudert** [comment]
No, we should use the latest GCC version. What is the issue with it?

**Comment by YoshitakaMo** [comment]
When using the latest gcc-9 compilers, an error occurred on the final linking step:

```
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/gromacs/2019.3 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DCMAKE_C_COMPILER=gcc-9 -DCMAKE_CXX_COMPILER=g++-9 -DGMX_MPI=OFF -DGMX_DOUBLE=OFF -DGMX_OPENMP=ON -DGMX_GPU=OFF -DBUILD_SHARED_LIBS=ON -DGMX_PREFER_STATIC_LIBS=OFF -DGMX_FFT_LIBRARY=FFTW3
-- The C compiler identification is GNU 9.1.0
-- The CXX compiler identification is GNU 9.1.0

...

[100%] Linking CXX shared library ../../lib/libgromacs.dylib
cd /tmp/gromacs-20190619-4451-axf8uo/gromacs-2019.3/build/src/gromacs && /usr/local/Cellar/cmake/3.14.5/bin/cmake -E cmake_link_script CMakeFiles/libgromacs.dir/link.txt --verbose=1
Undefined symbols for architecture x86_64:
  "_GOMP_loop_nonmonotonic_dynamic_next", referenced from:
      gmx_hbond(int, char**) [clone ._omp_fn.0] in gmx_hbond.cpp.o
  "_GOMP_loop_nonmonotonic_dynamic_start", referenced from:
      gmx_hbond(int, char**) [clone ._omp_fn.0] in gmx_hbond.cpp.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [lib/libgromacs.4.0.0.dylib] Error 1
make[1]: *** [src/gromacs/CMakeFiles/libgromacs.dir/all] Error 2
make: *** [all] Error 2
```

It didn't occur when using `gcc-8`, Apple LLVM, or the latest llvm 8.0.0. But Apple LLVM (on macOS) is not recommended because the compilation is encouraged to use `-fopenmp`.

**Comment by fxcoudert** [comment]
The error you quote is from `libomp` being installed on your system. This should not happen anymore with the latest version, can you update the formula to use GCC and I can help from there?

**Comment by YoshitakaMo** [comment]
Thank you for your help. That was my fault. After removing `libomp` of my `/usr/local/lib` which I manually installed, The compilation has successfully finished!

File: Formula/gromacs.rb [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
   depends_on "cmake" => :build
   depends_on "fftw"
-  depends_on "gsl"
+  depends_on "gcc" # for OpenMP
 
   def install
+    # Non-executable GMXRC files should be installed in DATADIR
     inreplace "scripts/CMakeLists.txt", "CMAKE_INSTALL_BINDIR",
                                         "CMAKE_INSTALL_DATADIR"
+    # fix an error on detecting CPU. see https://redmine.gromacs.org/issues/2927
```
Comment (fxcoudert) [inline Formula/gromacs.rb]: Upstream ticket says:

> I checked newer versions of CMake, for me the issue is gone

so we shouldn't hack that.

File: Formula/gromacs.rb [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+    inreplace "cmake/gmxDetectCpu.cmake",
+              "\"${GCC_INLINE_ASM_DEFINE} -I${PROJECT_SOURCE_DIR}/src -DGMX_CPUINFO_STANDALONE ${GMX_STDLIB_CXX_FLAGS} -DGMX_TARGET_X86=${GMX_TARGET_X86_VALUE}\")",
+              "${GCC_INLINE_ASM_DEFINE} -I${PROJECT_SOURCE_DIR}/src -DGMX_CPUINFO_STANDALONE ${GMX_STDLIB_CXX_FLAGS} -DGMX_TARGET_X86=${GMX_TARGET_X86_VALUE})"
+
+    args = std_cmake_args + %w[
+      -DCMAKE_C_COMPILER=gcc-9
+      -DCMAKE_CXX_COMPILER=g++-9
+      -DGMX_MPI=OFF
+      -DGMX_DOUBLE=OFF
```
Comment (fxcoudert) [inline Formula/gromacs.rb]: Why remove GSL? Why the other non-standard options?

**Review by YoshitakaMo — COMMENTED** [review]
I believe that `-DGMX_GSL` option is no longer used:
```
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    GMX_GSL
```
This option was deprecated three years ago ( https://redmine.gromacs.org/issues/1472 )

`-DGMX_OPENMP=ON`, `-DGMX_GPU=OFF`, `-DBUILD_SHARED_LIBS=ON`, `-DGMX_PREFER_STATIC_LIBS=OFF`, and `-DGMX_FFT_LIBRARY=FFTW3` options are now the default, so I removed them. 

Thank you for your considerate suggestions.

File: Formula/gromacs.rb [hunk 3]
```diff
@@ -1,27 +1,30 @@
 class Gromacs < Formula
   desc "Versatile package for molecular dynamics calculations"
   homepage "http://www.gromacs.org/"
-  url "https://ftp.gromacs.org/pub/gromacs/gromacs-2019.1.tar.gz"
-  sha256 "b2c37ed2fcd0e64c4efcabdc8ee581143986527192e6e647a197c76d9c4583ec"
-
-  bottle do
```
Comment (fxcoudert) [inline Formula/gromacs.rb]: Please leave the bottle block untouched.

File: Formula/gromacs.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/gromacs.rb]: OK. Please report it upstream, so they know it's not fixed and can investigate further.

File: Formula/gromacs.rb (hunk 3 above)
Comment (YoshitakaMo) [reply Formula/gromacs.rb]: I put the bottle block again.

**Comment by fxcoudert** [comment]
Thanks @YoshitakaMo

**Pull request closed without merging**